src/requirements/data/exams/ExamCredit.ts

```diff
@@ -1,235 +1,204 @@
 import { FWS_COURSE_ID } from '../constants';
 
-export type ExamRequirements = {
-  readonly name: string;
-  readonly fulfillment: {
-    readonly courseEquivalents?: Record<string, number[]>;
-    readonly minimumScore: number;
-    readonly credits: number;
-    readonly majorsExcluded?: string[];
-  };
+export type ExamFulfillment = {
+  readonly courseId: number;
+  readonly courseEquivalents?: Record<string, number[]>;
+  readonly minimumScore: number;
+  readonly credits: number;
+  readonly majorsExcluded?: string[];
 };
-export type ExamData = Record<'AP' | 'IB', ExamRequirements[]>;
+export type ExamFulfillments = Record<string, ExamFulfillment[]>;
+export type ExamData = Record<'AP' | 'IB', ExamFulfillments>;
 
 const examData: ExamData = {
-  AP: [
-    {
-      name: 'Biology',
-      fulfillment: {
-        minimumScore: 4,
-        credits: 4,
-      },
-    },
-    {
-      name: 'Biology',
-      fulfillment: {
-        minimumScore: 5,
-        credits: 8,
-      },
-    },
-    {
-      name: 'Chemistry',
-      fulfillment: {
+  AP: {
+    Biology: [
+      { courseId: 100, minimumScore: 4, credits: 4 },
+      { courseId: 101, minimumScore: 5, credits: 8 },
+    ],
+    Chemistry: [
+      {
+        courseId: 102,
         courseEquivalents: {
           DEFAULT: [351265], // CHEM 2070
           EN: [359187], // CHEM 2090
         },
```

Comment:
Yes, thanks for bringing this up! They should look similar because the conditions (in the requirements json) will be derived from the exam fulfillments (in the exam data).

The exam fulfillments data is currently being used for generating `courseTaken`s from `courseEquivalent`s, which was part of the old infra. For the new infra, it will be used to determine the conditions.

There are a few subtle differences, though:
- the exam fulfillments are properties of the exams, while the conditions are properties of the requirements
- `colleges` is a list of colleges, while the exam data uses the college as the key
- DEFAULT becomes a list of colleges

Here's a more concrete example if it helps. With the current data, AP CHEM has one equivalent course: CHEM 2090 for EN, and CHEM 2070 for DEFAULT (non-EN). This will cause the script to draw edges from an AP CHEM exam course with course id 102 to the requirements that CHEM 2070 *and* CHEM 2090 fulfill. This is the same thing as adding the course ids to the requirements json `courses` lists.
Say CHEM 2070 fulfills req1 and req2, and CHEM 2090 fulfills req1 and req3. Then, the exam course fulfills req1, req2, and req3.
- For req1, `conditions` will be undefined (or at least wouldn't have a key of 102) because CHEM 2070 and CHEM 2090 (all colleges) fulfill req1.
- For req2, `conditions` for course id 102 will be something like `colleges: [all colleges except 'EN']` because CHEM 2070 (non-EN colleges) fulfills req2.
- For req3, `conditions` for course id 102 will be something like `colleges: ['EN']` because CHEM 2090 (EN college only) fulfills req3.

Reply:
O i see so the checkers for defining which courses an exam fulfills are more complex than just defining the school

Reply:
Do you mean the checkers for defining which requirements an exam fulfills are more complex than just defining the course ids? If so, yes. We need the user data (college and major), which we can get at runtime but not when the requirements json is being generated.